src/main/java/uk/gov/companieshouse/accounts/association/controller/ControllerAdvisor.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.companieshouse.accounts.association.controller;
+
+import static uk.gov.companieshouse.accounts.association.utils.LoggingUtil.LOGGER;
+import static uk.gov.companieshouse.accounts.association.utils.ParsingUtil.parseJsonFrom;
+import static uk.gov.companieshouse.accounts.association.utils.RequestContextUtil.getXRequestId;
+import static uk.gov.companieshouse.accounts.association.utils.StaticPropertyUtil.APPLICATION_NAMESPACE;
+import static uk.gov.companieshouse.service.rest.err.Err.invalidPathBuilderWithLocation;
+import static uk.gov.companieshouse.service.rest.err.Err.serviceErrBuilder;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.validation.ConstraintViolationException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import uk.gov.companieshouse.accounts.association.exceptions.BadRequestRuntimeException;
+import uk.gov.companieshouse.accounts.association.exceptions.ForbiddenRuntimeException;
+import uk.gov.companieshouse.accounts.association.exceptions.InternalServerErrorRuntimeException;
+import uk.gov.companieshouse.accounts.association.exceptions.NotFoundRuntimeException;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+@ControllerAdvice
+public class ControllerAdvisor extends ResponseEntityExceptionHandler {
+
+    private <T extends Exception> Errors mapThrownExceptionsToErrors(final T exception, final HttpServletRequest request){
+        final var url = request.getRequestURL().toString();
+        final var queryParams = Objects.nonNull(request.getQueryString()) ? "?" + request.getQueryString() : "";
+        final var contextMap = new HashMap<String, Object>(Map.of("url", url, "query-parameters", queryParams));
```

Comment:
Hey don't think you need to use a HashMap here you can get around this by just explicitly stating the datatype rather than using the `var` keyword

```suggestion
        final Map<String, Object> contextMap = Map.of("url", url, "query-parameters", queryParams);
```